tests/integration/devfile/cmd_devfile_app_test.go

```diff
@@ -0,0 +1,109 @@
+package devfile
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var _ = Describe("odo devfile app command tests", func() {
+
+	var namespace, context, currentWorkingDirectory, originalKubeconfig string
+
+	// Using program command according to cliRunner in devfile
+	cliRunner := helper.GetCliRunner()
+
+	// This is run before every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		context = helper.CreateNewContext()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+		originalKubeconfig = os.Getenv("KUBECONFIG")
+		helper.LocalKubeconfigSet(context)
+		namespace = cliRunner.CreateRandNamespaceProject()
+		currentWorkingDirectory = helper.Getwd()
+	})
+
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		cliRunner.DeleteNamespaceProject(namespace)
+		helper.Chdir(currentWorkingDirectory)
+		err := os.Setenv("KUBECONFIG", originalKubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("when running help for app command", func() {
+		It("should display the help", func() {
+			appHelp := helper.CmdShouldPass("odo", "app", "-h")
+			Expect(appHelp).To(ContainSubstring("Performs application operations related to your project."))
```

Comment:
do you think we should add test for help? we always consciously change help messages.

Reply:
It was there in the s2i version of the test script, so I kept it.

Reply:
I think there is no need, what you say?

Reply:
removed as the test in the s2i version is sufficient.